Core/Core/Database/CoreData/Models/Submission.swift

```diff
@@ -145,11 +146,64 @@ extension Submission: WriteableModel {
             model.mediaComment = try MediaComment.save(mediaComment, in: client)
         }
 
-        SubmissionComment.save(item, in: client)
+        if item.submission_type != nil, item.submission_type != .none, item.submission_type != .not_graded, item.submission_type != .on_paper {
+            SubmissionComment.save(item, in: client)
```

Comment:
I'm not sure of the answer here so just posing this question, I didn't think we do a lot of `EntityName.save` because I thought most db operations were wrapped in some sort of persistence background thread block or db operation that then calls save at the end.  Do we really need this here?  Seems to follow it's own pattern versus the current pattern of the code as a whole.

Reply:
I think these methods are different than the overall `save` method that is called at the end of the db operation. These could be called something different than `save` to disambiguate them from the database client `save` method. These are more so `construct` methods or something to that affect.

Reply:
But I think the idea was that we want the models to know how to construct themselves rather than a use case or a parent model having to know how to construct their dependencies.

Reply:
I think this is the pattern of the code as a whole. A use case's write calls `Model.save`. When that model has sub-models, it recursively calls `Model.save` for those.

The problem here was that we created a comment for submission types that shouldn't show a comment. Rather than trying to filter it out later, we just don't make a comment entry for those submissions.

Reply:
I guess there is something unique here, since we create a `SubmissionComment` from an `APISubmission`. That is because we generally show each submission attempt in the comments view mixed in with the actual comments.

Reply:
Sounds good.  Thanks for helping me understand this part.